doc/faq.rst

```diff
@@ -8,10 +25,6 @@ Frequently Asked Questions
 
 Here we try to give some answers to questions that regularly pop up on the mailing list.
 
-.. contents:: Table of Contents
-   :local:
-   :depth: 2
-
```

Comment:
I think I liked having the top level recap of all the FAQ questions expanded:

![image](https://github.com/scikit-learn/scikit-learn/assets/89061/4f1956e1-4734-4164-a594-b94222bd5c01)

(in dev).

In the sphinx theme (+ this PR), we only have the partially expanded list of question in the dynamically expanded side panel:

![image](https://github.com/scikit-learn/scikit-learn/assets/89061/bb902171-31dd-4c61-b18c-6a99da7ec06c).

So it's much harder to see the list of all answered question at a glance, without having to scroll through the all document.